### ISSUE TITLE
možnost kontroly XML

### ISSUE THREAD

**Issue by pohadkar** [igloonet/mailkit-api#11]
je nějaká možnost vypsat si XML, které je vygenerováno a posláno do mailkitu?

**Comment by fejjo** [comment]
Knihovňa, zatiaľ nepodporuje túto možnosť, že si niekde definuješ debug mod. Všetky volania sa rovno posielajú na API mailkitu.

**Comment by fejjo** [comment]
Ak by si potreboval získať nejak to xmlko tak je možností viac ale najjednoduhšia je, že si to dumpnes priamo v tej metode kde sa to posiela. je to `XmlAdapter::getContent()` tak na riadku 137 máš premennú `$request` ktorú keď si dumpneš dostane priamo to XMLko čo sa posiela na api.

**Comment by pohadkar** [comment]
díky, vypsal jsem si to. ted musime zjistit u mailkitu, kde je chyba :)

**Comment by fejjo** [comment]
A co konkretne ti nefunguje ? Mozno ti poradim.

**Comment by pohadkar** [comment]
udelal jsem si takoveto zanorene asociativni pole:

$tvars['obj_adresa_fakturacni']['jmeno'] = 'xxxx';
$tvars['obj_adresa_fakturacni']['ulice'] = 'xxxx';
$tvars['obj_adresa_fakturacni']['mesto'] = 'xxxx';
$tvars['obj_adresa_fakturacni']['psc'] = 'xxxx';
$tvars['obj_adresa_dodaci']['jmeno'] = 'xxxx';
$tvars['obj_adresa_dodaci']['ulice'] = 'xxxx';
$tvars['obj_adresa_dodaci']['mesto'] = 'xxxx';
$tvars['obj_adresa_dodaci']['psc'] = 'xxxx';

vlozil jsem ho do template vars:
$message->setTemplateVars($tvars);

a mailkit ty vnorene jmeno ulice mesto psc nedal do mailu.

tak jsem chtel zkontrolovat, jestli v xml byly.

xml je vygenerovalo takto:


       <member>
        <name>obj_adresa_fakturacni</name>
        <value>
         <struct>
          <member>
           <name>jmeno</name>
           <value>
            <string>yyy</string>
           </value>
          </member>
          <member>
           <name>ulice</name>
           <value>
            <string>yyy</string>
           </value>
          </member>
          <member>
           <name>mesto</name>
           <value>
            <string>yyy</string>
           </value>
          </member>
          <member>
           <name>psc</name>
           <value>
            <string>yyy</string>
           </value>
          </member>
         </struct>
        </value>
       </member>
       <member>
        <name>obj_adresa_dodaci</name>
        <value>
         <struct>
          <member>
           <name>jmeno</name>
           <value>
            <string>yyy</string>
           </value>
          </member>
          <member>
           <name>ulice</name>
           <value>
            <string>yyy</string>
           </value>
          </member>
          <member>
           <name>mesto</name>
           <value>
            <string>yyy</string>
           </value>
          </member>
          <member>
           <name>psc</name>
           <value>
            <string>yyy</string>
           </value>
          </member>
         </struct>
        </value>
       </member>

coz mi prijde spravne :)
takze chyba musi byt na strane mailkitu. budem se jich ptat v cem je problem.

**Comment by fejjo** [comment]
Jo je mi to jasne :) Niesom si teraz isty, ci to takto pojde asi ak to nepotrebujes mat v tom poli tak by som si rovno vytvoril premenne obj_adresa_fakturacni_jmeno. Riesil som prave neco trochu podobne a to ze mas pole `orderItems` ktore obsahuje orderItem a to ma v sebe jednotlive info o produkte.
Co prave mne fungovalo ze v tempalte si si foreachol `orderItems` v tom si si includol nejaku sablonu ktora uz rovno odkazovala napr na `detailUrl`. Tak pridavam aspon tu strukturu ktora mi funguje na tento usecase. Mozno ti to pomoze a mozem ta poprosit potom sem hodit nieco co ti napisal mailkit a fungovalo to ? Diky


Ukazka obsahuje dva produkty:


      <member>
        <name>orderItems</name>
        <value>
         <array>
          <data>
           <value>
            <struct>
             <member>
              <name>detailUrl</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>orderThumbUrl</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>title</name>
              <value>
               <string>yy</string>
              </value>
             </member>
             <member>
              <name>unitPrice</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>unitPriceWithoutVat</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>amount</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>totalPrice</name>
              <value>
               <string>yy</string>
              </value>
             </member>
             <member>
              <name>totalPriceWithoutVat</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>params</name>
              <value>
               <array>
                <data/>
               </array>
              </value>
             </member>
             <member>
              <name>id</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>price</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>category</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>vatRate</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
            </struct>
           </value>
           <value>
            <struct>
             <member>
              <name>detailUrl</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>orderThumbUrl</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>title</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>unitPrice</name>
              <value>
               <string>yy</string>
              </value>
             </member>
             <member>
              <name>unitPriceWithoutVat</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>amount</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>totalPrice</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>totalPriceWithoutVat</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>params</name>
              <value>
               <array>
                <data/>
               </array>
              </value>
             </member>
             <member>
              <name>id</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>price</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>category</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
             <member>
              <name>vatRate</name>
              <value>
               <string>yyy</string>
              </value>
             </member>
            </struct>
           </value>
          </data>
         </array>
        </value>
       </member>

**Comment by pohadkar** [comment]
tak co se týče mailkitu, ukázalo se, že template vars 2 úrovně zanoření nedokáže převést z base64 na normální text :) jinak k těm proměnným se dostat dá a požít se dají (jen jsou zakódované např. QnJubw==). 
asi předpokládají, že ve 2 úrovni budou jen indexy a text bude zase až ve třetí úrovni zanoření.
takže buď zvolím to řešení, že dám obě adresy do 1 úrovně zanoření, nebo je hodím obě až do 3 úrovně. ještě budu testovat.

**Comment by fejjo** [comment]
Jo, dnes som sa to dozvedel tiez tak snad zajtra alebo do pondelka to upravim aj tu.
